### DISCUSSION TITLE
奇怪的大规模“吸血”现象

### DISCUSSION THREAD

**Discussion by pullingtab** [c0re100/qBittorrent-Enhanced-Edition#548] [General]
我注意到一件事。
似乎从23年的某个时间开始，出现了一些国内的吸血IP。
他们的特点是：

1. 使用相同的IP段，能在大量IP地之间切换，IP地址往往来自于内蒙古，辽宁，山东
2. 使用相同的端口
3. 反复下载少数几个文件
4. 这些IP会使用这样的peer ID：空ID，hp/，dt/，BitComet*，等

这群人有点奇怪。如果是单纯为了吸血，没道理总是吸同一个文件呀。
事实上，他们似乎根本不是在吸血，如果你真的连上他们，他们会给你发数据：
![奇怪的BT客户端](https://github.com/c0re100/qBittorrent-Enhanced-Edition/assets/8113380/b43272e1-ff38-42b8-a92e-a1ce6580786f)
客观上，他们的行为有利于提高这些资源的下载速度。
当然这是在你没有下完这个文件的情况下。如果你有了他们感兴趣的文件，并且好心的刻意做种，你的设备就会被大量的客户端连上，挤满你的上传速度。
![我下完了就开始吸我](https://github.com/c0re100/qBittorrent-Enhanced-Edition/assets/8113380/8006f32c-0569-42e1-b87f-a803a78eea5b)
这种行为不能称为恶意的吸血，客观上他们只是在浪费你的上传带宽。他们不仅没有危害P2P网络的价值，相反还让某些文件的下载更快了。

有懂行的么？**他们这是在干啥？**带宽不要钱的么？
而且从他们的行为来看，是不是**没有必要完全封禁他们**？我们只需要想办法设置一个规则，给这些客户端在做种期间狂吸我们的客户端限个速，世界就会更美好。是这样么？

hp*客户端似乎不会给你上传数据：
![image](https://github.com/c0re100/qBittorrent-Enhanced-Edition/assets/8113380/b7c9b14c-593e-48ba-8234-11fffb401f87)

**Comment by machine-doll** [comment]
这些是为了消耗下载流量用的，用来平衡下载和上传的流量，规避运营商对于PCDN的检测规则  
[来自NGA的这个讨论](https://g.nga.cn/read.php?tid=39822117&rand=48)告诉你了为什么会有这种客户端，它们并不是为了得到种子里面的数据，只是单纯地为了消耗流量，和以往的“吸血”完全不一样

**Comment by lzhzh1** [comment]
事实上，仅有BC0204会做种，因为它们是功能正常的比特彗星客户端，不是经未知修改的HP或DT客户端。
（可以看到图中有比特彗星特有的长效种子p2sp功能）
![1](https://github.com/c0re100/qBittorrent-Enhanced-Edition/assets/117511260/9da4997a-cc90-42ad-8415-1c1d8b4c2c88)
很可惜的是，现在绝大部分都是HP客户端在吸血。
当然，在之前的一个[issue](https://github.com/c0re100/qBittorrent-Enhanced-Edition/issues/540#issuecomment-2053675258)（它应该放在discussions里）中，我没有经过观察主观臆断BC0204为伪装客户端，在此表示抱歉...
不过你也应该理解为什么会有人屏蔽它们：
能关注到这些事，并且来github上寻求解决方法的用户，大部分都是有大量做种任务的。若是上传速度都被一个或几个任务中**循环下载**的BC0204占据，其他资源的上传效率将大幅度降低。既然它有循环下载的异常行为，屏蔽它也是无可厚非的。
就我而言，我希望我少有的上传能贡献给广泛且真实的人，而不是反复下载同几个资源的机器。
相关讨论：[滥用比特彗星进行循环下载](https://www.cometbbs.com/t/%E5%90%B8%E8%A1%80%E5%AE%A2%E6%88%B7%E7%AB%AF%E6%94%B6%E9%9B%86%E4%B8%8E%E8%A7%A3%E5%86%B3%E6%96%B9%E6%B3%95/92122/31?u=lzhzh_1)

**Reply by CalunVier** [reply]
问个题外话。其他客户端可以连接到p2sp吗？有没有相关插件？

**Reply by lzhzh1** [reply]
> 问个题外话。其他客户端可以连接到p2sp吗？有没有相关插件？

比特彗星的p2sp种源目前应该只有同为比特彗星的客户端能连接

**Reply by CalunVier** [reply]
> > 问个题外话。其他客户端可以连接到p2sp吗？有没有相关插件？
> 
> 比特彗星的p2sp种源目前应该只有同为比特彗星的客户端能连接

真是太糟糕了，比特彗星并不是”最好用“的客户端，但是长效种子网络看起来真的十分诱人。为什么没有其他客户端支持？是否有可能通过插件或其他方法让其他客户端支持？您对此有何见解？

**Reply by lzhzh1** [reply]
> > > 问个题外话。其他客户端可以连接到p2sp吗？有没有相关插件？
> > 
> > 
> > 比特彗星的p2sp种源目前应该只有同为比特彗星的客户端能连接
> 
> 真是太糟糕了，比特彗星并不是”最好用“的客户端，但是长效种子网络看起来真的十分诱人。为什么没有其他客户端支持？是否有可能通过插件或其他方法让其他客户端支持？您对此有何见解？

长效种子需要有中心服务器支持才能够搜寻到种源，这应该是大多数客户端没有p2sp功能的原因。
这里有一篇以前的讨论帖，比特彗星开发者wxhere15回复没办法让qB连到长效种子。我不知道这是不是说明比特彗星的长效种子只允许比特彗星连接，而没办法通过其他方式连接...
[有办法让qb连上比特彗星的长效种子吗](https://www.cometbbs.com/t/%E6%9C%89%E5%8A%9E%E6%B3%95%E8%AE%A9qb%E8%BF%9E%E4%B8%8A%E6%AF%94%E7%89%B9%E5%BD%97%E6%98%9F%E7%9A%84%E9%95%BF%E6%95%88%E7%A7%8D%E5%AD%90%E5%90%97/59597)
